### ISSUE TITLE
[BUG] Getting NaT (Missing timestamp) when trying to convert frequency of data set

### ISSUE THREAD

**Issue by lucharo** [autogluon/autogluon#4917]
**Bug Report Checklist**
I am trying to recreate the advanced timeseries tutorial, I am doing some demand forecast with missing values. 

I load my data into a `TimeSeriesDataFrame` with columns "id", "date_actual" (daily frequency) and "value" (my target) then I am trying to convert to daily frequency and fill in missing values as explained in the website (https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html#handling-irregular-data-and-missing-values) but instead I get NaT in the timestamp but with some values assigned , I do get NaN in some columns though, not sure what's going on: 

**Screenshots / Logs**

<img width="469" alt="Image" src="https://github.com/user-attachments/assets/e4f076f7-f70a-4a3d-b2a8-dc847b9dfa21" />

**Installed Versions**

<details>

INSTALLED VERSIONS
------------------
date                   : 2025-02-21
time                   : 12:19:09.945868
python                 : 3.12.0.final.0
OS                     : Darwin
OS-release             : 24.3.0
Version                : Darwin Kernel Version 24.3.0: Thu Jan  2 20:24:23 PST 2025; root:xnu-11215.81.4~3/RELEASE_ARM64_T8122
machine                : arm64
processor              : arm
num_cores              : 8
cpu_ram_mb             : 8192.0
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 60153

accelerate             : 0.34.2
autogluon              : 1.2
autogluon.common       : 1.2
autogluon.core         : 1.2
autogluon.features     : 1.2
autogluon.multimodal   : 1.2
autogluon.tabular      : 1.2
autogluon.timeseries   : 1.2
boto3                  : 1.36.25
catboost               : 1.2.7
coreforecast           : 0.0.12
defusedxml             : 0.7.1
einops                 : 0.8.1
evaluate               : 0.4.3
fastai                 : 2.7.18
fugue                  : 0.9.1
gluonts                : 0.16.0
huggingface-hub        : 0.29.1
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.5
joblib                 : 1.4.2
jsonschema             : 4.21.1
lightgbm               : 4.5.0
lightning              : 2.5.0.post0
matplotlib             : 3.10.0
mlforecast             : 0.13.4
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.8.1
numpy                  : 1.26.4
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.2.3
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : None
optimum-intel          : None
orjson                 : 3.10.15
pandas                 : 2.2.3
pdf2image              : 1.17.0
Pillow                 : 11.1.0
psutil                 : 6.1.1
pyarrow                : 19.0.1
pytesseract            : 0.3.10
pytorch-lightning      : 2.5.0.post0
pytorch-metric-learning: 2.3.0
ray                    : 2.39.0
requests               : 2.32.3
scikit-image           : 0.24.0
scikit-learn           : 1.5.2
scikit-learn-intelex   : None
scipy                  : 1.15.2
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.7.5
statsforecast          : 1.7.8
tabpfn                 : None
tensorboard            : 2.19.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.5.1
torchmetrics           : 1.2.1
torchvision            : 0.20.1
tqdm                   : 4.67.1
transformers           : 4.49.0
utilsforecast          : 0.2.4
vowpalwabbit           : None
xgboost                : 2.1.4

</details>

**Comment by lucharo** [comment]
I had my date/timestamp column as type datetime64[ns] and it was exhibiting this behaviour and when I changed to object, it worked, a bit counterintuitive, is it meant to work like this?

I added this line:

```python
train["date_actual"] = pd.to_datetime(train.date_actual).dt.strftime('%Y-%m-%d').astype('object')

```

**Comment by shchur** [comment]
Hi @lucharo, thanks for reporting this problem. Is there any chance that you could share the full raw values in the `item_id` and `timestamp` columns of your dataset (anonymized item_ids should be fine too)? Something strange seems to be happening there and AutoGluon should at least produce meaningful warnings when NaT entries are encountered. However, I'm not sure how to reproduce this issue locally.

**Comment by lucharo** [comment]
Hey @shchur, thank you for the follow up. I'll give the full context and hopefully we get to a reprex: 

I initially load the data into a polars data frame (from a postgres db), at this stage the polars columns and types are:
```python
['id', 'date_actual', 'value'] [String, Date, Float64]
```

then when I turn this df into pandas I get:
```python
id                     object
date_actual    datetime64[ms]
value                 float64
dtype: object
```

If i don't do anything to these dates, I get the error I showed you earlier, instead I need to fix it by running:

```python
train_df["date_actual"] = pd.to_datetime(train_df.date_actual).dt.strftime('%Y-%m-%d').astype('object')
```

where `train_df` is my pandas df

---

another quirky thing I have found when exporting the data is that if I save the data to a csv and reload it again via pandas `pd.read_csv` I get the date column of type `object`, what autogluon expects. 

the quirk here is in the polars to pandas type conversion and then autogluon or pandas functions now working on the datetime64[ms] type column:

I think this should do as a reprex:

```python
# reprex.py

# First, let's create a simple toy dataset with 5 rows and 3 columns
import pandas as pd
import polars as pl
import csv
from datetime import datetime, timedelta
from autogluon.timeseries import TimeSeriesDataFrame, TimeSeriesPredictor

# Creating the data
toy = [
    ["H1", "2023-01-15", 42],
    ["H2", "2023-02-20", 78],
    ["H1", "2023-03-10", 33],
    ["H2", "2023-04-05", 91],
    ["H1", "2023-05-22", 67]
]

# Saving as CSV
with open('toy_dataset.csv', 'w', newline='') as file:
    writer = csv.writer(file)
    writer.writerow(['item_id', 'timestamp', 'value'])  # Header
    writer.writerows(toy)

print("CSV file created: toy_dataset.csv")

# Loading with Polars using try_parse_dates=True
pl_df = pl.read_csv('toy_dataset.csv', try_parse_dates=True)
print("\nPolars from CSV DataFrame:")
print(pl_df.dtypes)

# Converting to Pandas from polars
pd_df = pl_df.to_pandas()
print("\nPandas from Polars DataFrame Info:")
print(pd_df.dtypes)

# Loading into pandas directly:
pd_df_direct = pd.read_csv("toy_dataset.csv")
print("\nPandas from CSV DataFrame (Info:")
print(pd_df_direct.dtypes)

for df in [pd_df, pd_df_direct]:
    print(TimeSeriesDataFrame.from_data_frame(
        df
    ).sort_index().convert_frequency("D").fill_missing_values(method="constant").head())
```

you can run it with:
```sh
UV_PYTHON=3.11 uv --with autogluon --with polars reprex.py  
```
